feat: Add settings dialog

Mostafa-alsaygh opened the pull request:
# What does this PR do?

Adds a new settings dialog with options to control sound and music volume. This includes the necessary layout, drawables, and string resources.

<img width="637" height="643" alt="image" src="https://github.com/user-attachments/assets/2bff77d0-87ed-471a-a927-0db1f04673c5" />

Mostafa-alsaygh committed:
feat: Add settings dialog

Adds a new settings dialog with options to control sound and music volume. This includes the necessary layout, drawables, and string resources.

Mostafa-alsaygh committed:
style: Refactor and restyle settings dialog

fatmahgazy commented on app/src/main/res/layout/settings_dialog.xml:
```diff
@@ -0,0 +1,162 @@
+        android:paddingBottom="8dp"
+        android:text="@string/settings_dialog"
+        android:textColor="#FFFFFF"
+        android:textSize="18sp"
```
I think you can use style here

fatmahgazy commented on gradle/libs.versions.toml:
```diff
@@ -1,5 +1,5 @@
 [versions]
-agp = "8.13.0"
+agp = "8.11.0"
```
why you change version?

Mostafa-alsaygh replied:
It dont run in my device... I will return it back my mistake

Mostafa-alsaygh committed:
feat: Refactor settings dialog UI

Mostafa-alsaygh committed:
feat: Update settings dialog seekbar style

Mostafa-alsaygh committed:
style: adjust settings dialog layout

Mostafa-alsaygh committed:
Merge branch 'develop' into feat/settings_dialog

# Conflicts:
#	app/src/main/res/values/strings.xml

Mostafa-alsaygh committed:
feat: Improve visual style of divider

Mostafa-alsaygh committed:
build: return AGP to 8.13.0

Abdulrahman-Ragab-01 commented on app/src/main/res/drawable/button_discard_background.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
please use the button style that is already exist in the drawable as `secondary_button_background` for the button background and its text color `secondary_button_text_color`

Abdulrahman-Ragab-01 commented on app/src/main/res/drawable/button_save_background.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
please use the button style that is already exist in the drawable as `primary_button_background` for the button background and its text color `primary_button_text_color`

Abdulrahman-Ragab-01 commented on app/src/main/res/drawable/dialog_background.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
please use the `corners_cut_shape` as a background for the dialog that is already exist to make the corners style as in figma

Abdulrahman-Ragab-01 commented on app/src/main/res/layout/settings_dialog.xml:
```diff
@@ -0,0 +1,175 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
please use constraint layout instead of frame layout

Mostafa-alsaygh replied:
why??!

Mostafa-alsaygh replied:
i made it constraint layout for the close item, it did not work with me other that

Abdulrahman-Ragab-01 replied:
A single constraint layout is better for performance, all the dialog can be build with that constraint layout and handling the chains

Mostafa-alsaygh replied:
how to handle the out title item ?? see the pic

Abdulrahman-Ragab-01 commented on app/src/main/res/layout/settings_dialog.xml:
```diff
@@ -0,0 +1,175 @@
+    android:clipToPadding="false"
+    android:paddingTop="20dp">
+
+    <RelativeLayout
```
you can avoid using any other layouts and only depend on the parent constraint layout

Mostafa-alsaygh replied:
it is needed for the outside title see the pic

Mostafa-alsaygh replied:
constraint layout is the optional for the cards and or items

Abdulrahman-Ragab-01 replied:
you can handle the inner components by using chains

Mostafa-alsaygh replied:
ok i will see it

Abdulrahman-Ragab-01 commented on app/src/main/res/layout/settings_dialog.xml:
```diff
@@ -0,0 +1,175 @@
+            android:src="@android:drawable/ic_menu_close_clear_cancel"
+            app:tint="#FFFFFF" />
+
+        <LinearLayout
```
you can avoid using any other layouts and only depend on the parent constraint layout

Mostafa-alsaygh replied:
i need them cause we have a layout_weight and it is ok to leave it

Mostafa-khaled21 replied:
@Mostafa-alsaygh  I think you can do it with chains

Abdulrahman-Ragab-01 replied:
that is not the best approach because the performance will be affected with many layouts, you can handle the inner components by using chains

Mostafa-alsaygh replied:
ok i will see it guys

Abdulrahman-Ragab-01 commented on app/src/main/res/layout/settings_dialog.xml:
```diff
@@ -0,0 +1,175 @@
+                android:thumbTint="@null" />
+        </LinearLayout>
+
+        <LinearLayout
```
you can avoid using any other layouts and only depend on the parent constraint layout

Abdulrahman-Ragab-01 commented on app/src/main/res/layout/settings_dialog.xml:
```diff
@@ -0,0 +1,175 @@
+            android:layout_marginTop="30dp"
+            android:background="@drawable/background_divider" />
+
+        <LinearLayout
```
you can avoid using any other layouts and only depend on the parent constraint layout

Abdulrahman-Ragab-01 commented on app/src/main/res/layout/settings_dialog.xml:
```diff
@@ -0,0 +1,175 @@
+                android:textAllCaps="false"
+                android:textColor="@color/on_primary"
+                android:textSize="14sp" />
+
```
in those two buttons you can use our custom buttons style for the buttons background and colors as I recommend above

Mostafa-alsaygh replied:
when i remove the it turn black

Abdulrahman-Ragab-01 commented on app/src/main/res/values-night/strings.xml:
```diff
@@ -3,4 +3,12 @@
     <string name="day_fire">day fire</string>
+    <string name="close">Close</string>
+    <string name="sound">Sound</string>
+    <string name="sound_setting">Sound</string>
```
that string is duplicated with the one above it, please delete one of them

Abdulrahman-Ragab-01 commented on app/src/main/res/values-night/strings.xml:
```diff
@@ -3,4 +3,12 @@
+    <string name="sound">Sound</string>
+    <string name="sound_setting">Sound</string>
+    <string name="music_setting">Music</string>
+    <string name="musics">Music</string>
```
that string is duplicated with the one above it, please delete one of them

Abdulrahman-Ragab-01 commented on app/src/main/res/values/strings.xml:
```diff
@@ -76,6 +76,14 @@
     <string name="achievement_image">achievement image</string>
+    <string name="close">Close</string>
+    <string name="sound">Sound</string>
+    <string name="sound_setting">Sound</string>
```
that string is duplicated with the one above it, please delete one of them

Abdulrahman-Ragab-01 commented on app/src/main/res/values/strings.xml:
```diff
@@ -76,6 +76,14 @@
+    <string name="sound">Sound</string>
+    <string name="sound_setting">Sound</string>
+    <string name="music_setting">Music</string>
+    <string name="musics">Music</string>
```
that string is duplicated with the one above it, please delete one of them

Abdulrahman-Ragab-01 commented on app/src/main/res/drawable/settings_dialog_background.xml:
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
that image for the title of the dialog is already exist in the develop as `heptagon_shape` you can use it

Abdulrahman-Ragab-01 commented on app/src/main/res/layout/settings_dialog.xml:
that dialog has two views like blue line beside the dialog title in the right and left sides use put them, here is its implementation

```
<View
        android:id="@+id/gradient_left"
        android:layout_width="0dp"
        android:layout_height="1dp"
        app:layout_constraintDimensionRatio="56:1"
        android:background="@drawable/blue_linear_gradiant"
        android:rotation="180"
        android:translationX="5dp"
        app:layout_constraintEnd_toStartOf="@id/heptagon_shape"
        app:layout_constraintTop_toTopOf="@id/corners_cut_shape" />

    <View
        android:id="@+id/gradient_right"
        android:layout_width="0dp"
        android:layout_height="1dp"
        app:layout_constraintDimensionRatio="56:1"
        android:background="@drawable/blue_linear_gradiant"
        android:translationX="-5dp"
        app:layout_constraintStart_toEndOf="@id/heptagon_shape"
        app:layout_constraintTop_toTopOf="@id/corners_cut_shape" />
```

Abdulrahman-Ragab-01 commented on app/src/main/res/layout/settings_dialog.xml:
```diff
@@ -0,0 +1,175 @@
+        android:background="@drawable/dialog_background"
+        android:padding="20dp">
+
+        <ImageButton
```
you can use the close icon that is already exist, here is an example for its usage

```
    <ImageView
        android:id="@+id/close_shape"
        android:layout_width="0dp"
        android:layout_height="26dp"
        android:clickable="true"
        android:contentDescription="@string/card_background"
        android:src="@drawable/close"
        app:layout_constraintDimensionRatio="22:26"
        app:layout_constraintEnd_toEndOf="@+id/corners_cut_shape"
        app:layout_constraintTop_toTopOf="@+id/corners_cut_shape" />
```

Mostafa-alsaygh replied:
thanks

Abdulrahman-Ragab-01 reviewed (changes requested):
great job, Mostafa 👏🏻
just take a look in the comments please

london-bot[bot] committed:
Branch was auto-updated.

Mostafa-alsaygh committed:
refactor: Update settings dialog UI

This commit refactors the settings dialog UI.

Key changes:
- Replaced the FrameLayout and RelativeLayout with a single ConstraintLayout for a more efficient layout structure.
- Updated the dialog's title and button styles for a new visual design.
- Removed unused drawable and string resources.

fatmahgazy commented:
> # What does this PR do?
> Adds a new settings dialog with options to control sound and music volume. This includes the necessary layout, drawables, and string resources.
> 
> <img alt="image" width="578" height="528" src="https://private-user-images.githubusercontent.com/106277855/500159450-389b937c-bfd0-4178-bf89-d9f76926d9b6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA2MjIyNzIsIm5iZiI6MTc2MDYyMTk3MiwicGF0aCI6Ii8xMDYyNzc4NTUvNTAwMTU5NDUwLTM4OWI5MzdjLWJmZDAtNDE3OC1iZjg5LWQ5Zjc2OTI2ZDliNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAxNlQxMzM5MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYzgwOWE3ZDkyZDRmOTdkMTAxZTM0OTA3OWM3ODM3NTc0ZjVlMmUyNGMwMzg4Zjg2NDViYjU1YTMxMWRkZWNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.9H9odpfYhQaV9jxIamI3qm6vyYImKP1vLqbK28gyHVk">

**This is not the same desgin in figma , can you check it please** 

<img width="1001" height="645" alt="image" src="https://github.com/user-attachments/assets/d20481a2-23d6-454c-b1c0-2cc44eda70ce" />

Mostafa-alsaygh committed:
refactor: Restructure settings dialog layout

Mostafa-alsaygh committed:
refactor: Remove dialog background drawable

Abdulrahman-Ragab-01 reviewed (approved):
Great Job 👏🏻❤️
